lib/pos_mempool_manager.go

```diff
@@ -0,0 +1,50 @@
+package lib
```

Comment:
Holding off on reviewing this file since it relies on the block producer.

Reply:
The time has come